analyzer/runtime/extract.go

```diff
@@ -808,7 +808,9 @@ func extractEvents(blockData *BlockData, relatedAccountAddresses map[apiTypes.Ad
 						eventData.RelatedAddresses[approvedAddr] = true
 					}
 					if _, ok := blockData.PossibleTokens[eventAddr]; !ok {
-						blockData.PossibleTokens[eventAddr] = &evm.EVMPossibleToken{}
+						blockData.PossibleTokens[eventAddr] = &evm.EVMPossibleToken{
+							TotalSupplyChange: &big.Int{},
```

Comment:
This creates/encourages long-range dependencies: All the code that initializes values in `PossibleTokens`, present and future, needs to be aware of [the code that expects `.TotalSupplyChange` to be non-nil](https://github.com/oasisprotocol/nexus/blob/43a480c419552aaf6d6499139e3b79f6fda54e22/analyzer/runtime/extract.go#L764-L764). We're bound to mess up again, sooner or later.

I suggest one of these alternative fixes:
- Make the code linked above more robust, i.e. make it initialize `.TotalSupplyChange` if it's `nil`.
- Change the type of `.TotalSupplyChange` so it's `big.Int`, not `*big.Int`. This will prevent future nilness problems for good.

Reply:
> Change the type of .TotalSupplyChange so it's big.Int, not *big.Int

Slightly prefer this because it simplifies the other parts as well; going with this

Reply:
Thanks Andy.

:facepalm: to the `big.Int` API. I guess give how/where we use `totalSupplyChange`, we could have gone with `common.BigInt`, where API is either more sane or we're free to make it more sane. Oh well, enough bikeshedding I guess.

Reply:
> big.Int API

TIL that even though `big.Int`'s [methods are pointer receiver](https://pkg.go.dev/math/big#Int.Add), it doesn't matter if it's in a struct field like `totalSupply big.Int` ¯\_(ツ)_/¯